src/transformers/models/jais2/modular_jais2.py

```diff
@@ -0,0 +1,238 @@
+        initializer_range: Optional[float] = 0.02,
+        layer_norm_eps: Optional[float] = 1e-5,
+        use_cache: Optional[bool] = True,
+        pad_token_id: Optional[int] = None,
+        bos_token_id: Optional[int] = 0,
+        eos_token_id: Optional[int] = 150024,
+        pretraining_tp: Optional[int] = 1,
+        tie_word_embeddings: Optional[bool] = False,
+        attention_bias: Optional[bool] = True,
+        attention_dropout: Optional[float] = 0.0,
+        mlp_bias: Optional[bool] = True,
+        head_dim: Optional[int] = None,
+        rope_theta: Optional[float] = 500000.0,
+        rope_parameters: Optional[RopeParameters | dict[str, RopeParameters]] = None,
+        **kwargs,
+    ):
+        # If rope_parameters not provided, create default with rope_theta
+        if rope_parameters is None:
+            rope_parameters = RopeParameters(rope_theta=rope_theta)
+
+        # Define rms_norm_eps for the parent init to use
+        rms_norm_eps = layer_norm_eps
+
+        super().__init__(
+            vocab_size=vocab_size,
+            hidden_size=hidden_size,
+            intermediate_size=intermediate_size,
+            num_hidden_layers=num_hidden_layers,
+            num_attention_heads=num_attention_heads,
+            num_key_value_heads=num_key_value_heads,
+            hidden_act=hidden_act,
+            max_position_embeddings=max_position_embeddings,
+            initializer_range=initializer_range,
+            rms_norm_eps=rms_norm_eps,
+            use_cache=use_cache,
+            pad_token_id=pad_token_id,
+            bos_token_id=bos_token_id,
+            eos_token_id=eos_token_id,
+            pretraining_tp=pretraining_tp,
+            tie_word_embeddings=tie_word_embeddings,
+            attention_bias=attention_bias,
+            attention_dropout=attention_dropout,
+            mlp_bias=mlp_bias,
+            head_dim=head_dim,
+            rope_parameters=rope_parameters,
+            **kwargs,
+        )
+        # Rename the attribute from rms_norm_eps to layer_norm_eps
+        self.layer_norm_eps = self.rms_norm_eps
+
+        # Validate and standardize RoPE parameters
+        self.standardize_rope_params()
+        self.validate_rope()
+
+
+__all__ = ["Jais2Config"]
+
+
+class Jais2MLP(NemotronMLP):
+    pass
+
+
+class Jais2DecoderLayer(LlamaDecoderLayer):
+    def __init__(self, config: Jais2Config, layer_idx: int):
+        super().__init__(config, layer_idx)
+        self.input_layernorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.post_attention_layernorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.mlp = Jais2MLP(config)
+
+
+class Jais2PreTrainedModel(LlamaPreTrainedModel):
+    config_class = Jais2Config
+    base_model_prefix = "model"
+    _no_split_modules = ["Jais2DecoderLayer"]
+    _skip_keys_device_placement = "past_key_values"
+    _supports_flash_attn_2 = True
```

Comment:
```suggestion
    pass
```
We should inherit everything correctly. Side note several of these are outdated, e.g. `_supports_flash_attn_2` is now under `_supports_flash_attn` included